custom-war-packager-lib/src/main/java/io/jenkins/tools/warpackager/lib/util/CollectionsHelper.java

```diff
@@ -0,0 +1,30 @@
+package io.jenkins.tools.warpackager.lib.util;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * @author Oleg Nenashev
+ * @since TODO
+ */
+public class CollectionsHelper {
+
+    @CheckForNull
+    public static <K,V> V putIfNotNull(@Nonnull Map<K, V> dest, @Nonnull K key, @CheckForNull V value) {
+        if (value != null) {
+            return dest.put(key, value);
+        }
+        return dest.get(key);
+    }
+
+    @CheckForNull
+    public static <K,V> V getOrFail(@Nonnull Map<K, V> src, @Nonnull K key) throws IOException {
+        V value = src.get(key);
+        if (value == null) {
+            throw new IOException("Cannot mandatory key " + key);
```

Comment:
This message is hardly descriptive of the error... 
NIT: I would say that it should throw an IllegalArgumentException instead of an IOException, but that is being picky :P

Reply:
Yes, I didn't do good job on reporting here